services/records_to_socrata.py

```diff
@@ -38,7 +38,23 @@ def bools_to_strings(records):
         for k, v in record.items():
             if isinstance(v, bool):
                 record[k] = str(v)
-    return records
+
+
+def remove_unknown_fields(record_fieldNames, payload, client_metadata):
+    """
+    Modifies payload by removing the fields not found in Socrata
+    Prevents "400 Client Error: Bad Request. Illegal field name sent" response
+    :param record_fieldNames: list of fieldNames in knack dataset
+    :param payload: records payload to send to Socrata
+    :param client_metadata: Socrata metadata for app
+    """
+    column_names = [c["fieldName"] for c in client_metadata["columns"]]
+    unknown_fields = [fieldName.lower() for fieldName in record_fieldNames if fieldName.lower() not in column_names]
+    if len(unknown_fields) > 0:
```

Comment:
this is fine, but more Pythonic would be simply `if unknown_fields:`. Empty lists and dicts are falsey in Python.